soroban-sdk/src/crypto.rs

```diff
@@ -189,6 +197,28 @@ impl Crypto {
     pub fn bn254(&self) -> bn254::Bn254 {
         bn254::Bn254::new(self.env())
     }
+
+    /// Performs a Poseidon hash using a sponge construction
+    pub fn poseidon_hash(&self, inputs: &Vec<U256>, field: Symbol) -> U256 {
+        // The initial value for the capacity element initialized with 0 for standard Poseidon
+        let iv = U256::from_u32(&self.env, 0);
+        let mut sponge = PoseidonSponge::new(&self.env, iv, field);
+        for input in inputs.iter() {
+            sponge.absorb(input);
+        }
+        sponge.squeeze()
+    }
+
+    /// Performs a poseidon2 hash with a sponge construction equivalent to the one in the Barretenberg proving system
+    pub fn poseidon2_hash(&self, inputs: &Vec<U256>, field: Symbol) -> U256 {
```

Comment:
Would it be appropriate to name these fns simple `poseidon` and `poseidon2`? Looking at the other functions in here we do not typically suffix hashing functions with `_hash`.

Or are there other things you can do with poseidon that would result in us needing to distinguish a hash from something else?